iris-mpc-gpu/src/server/actor.rs

```diff
@@ -1060,6 +1112,144 @@ impl ServerActor {
             Eye::Right => (&self.db_match_list_right, &self.batch_match_list_right),
         };
 
+        let (
+            match_distances_buffers_codes,
+            match_distances_buffers_masks,
+            match_distances_counters,
+            match_distances_indices,
+        ) = match eye_db {
+            Eye::Left => (
+                &self.match_distances_buffer_codes_left,
+                &self.match_distances_buffer_masks_left,
+                &self.match_distances_counter_left,
+                &self.match_distances_indices_left,
+            ),
+            Eye::Right => (
+                &self.match_distances_buffer_codes_right,
+                &self.match_distances_buffer_masks_right,
+                &self.match_distances_counter_right,
+                &self.match_distances_indices_right,
+            ),
+        };
+
+        let mut total_distance_counter = 0;
+        for (i, device) in self.device_manager.devices().iter().enumerate() {
+            let counter = device.dtoh_sync_copy(&match_distances_counters[i]).unwrap();
+            total_distance_counter += counter[0];
+        }
+
+        tracing::info!("Matching distances collected: {}", total_distance_counter);
+
+        if total_distance_counter >= MATCH_DISTANCES_BUFFER_SIZE as u32 {
```

Comment:
Meta: Is the buffer size:
- small enough to provide frequent enough bucket stats
- large enough to be private enough

I feel the answer to 2 is yes, for 1 I do not know.

Reply:
yeah also not quite sure yet, let me get some numbers on "the average amount of partial matches per signup". Then we can estimate how long it takes.
FYI: also the number of buckets is more of a placeholder still, I think a bit higher resolution will be required.